### PR TITLE
Misc: More warnings fixup.

### PR DESCRIPTION
**Pull request by lightningterror**
### Description of Changes

Compiler warnings cleanup.
### Rationale behind Changes

Warnings cleanup.
### Suggested Testing Steps

CI passes.

### CODE REVIEW

File: pcsx2/R3000A.cpp [hunk 1]
```diff
@@ -222,7 +222,7 @@ __ri void iopEventTest()
 	{
 		// start the next branch at the next counter event by default
 		// the interrupt code below will assign nearer branches if needed.
-		if (psxNextCounter < (psxRegs.iopNextEventCycle - psxNextsCounter))
+		if (static_cast<u32>(psxNextCounter) < (psxRegs.iopNextEventCycle - psxNextsCounter))
```
Comment (stenzek) [inline pcsx2/R3000A.cpp:225]: This one's a bit suss, I think the right side should be casted to signed instead (it's saying is the delta less than the time to the next event).

Current behaviour w/ the warning is to do an unsigned comparison, but I'm not sure if that's correct.

Comment (refractionpcsx2) [reply pcsx2/R3000A.cpp:225]: yeah, in the event it's overrun, you don't want it waiting 2 billion cycles :)

Comment (lightningterror) [reply pcsx2/R3000A.cpp]: All good now?

Comment (refractionpcsx2) [reply pcsx2/R3000A.cpp]: Yes that's better

Comment (lightningterror) [reply pcsx2/R3000A.cpp]: Was more of just in case it's not big enough, but sure can switch to s32.

Comment (refractionpcsx2) [reply pcsx2/R3000A.cpp]: Well the others are only u32 so it can never be bigger than 32bit

**Pull request merged**